enterprise/backend/test/metabase_enterprise/advanced_permissions/common_test.clj

```diff
@@ -704,47 +704,53 @@
+            (upload-test/with-upload-table!
+              [table-b (upload-test/create-upload-table! :schema-name schema-name)]
+              (let [db-id       (u/the-id (mt/db))
+                    append-csv! #(upload-test/update-csv-with-defaults!
+                                  action
+                                  :table-id (:id table-a)
+                                  :user-id (mt/user->id :rasta))]
+                (doseq [[schema-perms can-append? test-string]
+                        [[:all true "Data permissions on schema should succeed"]
+                         [:none false "No permissions on schema should fail"]
+                         [{(:id table-a) :all} true "Data permissions on table should succeed"]
+                         [{(:id table-b) :all} false "Data permissions only on another table in the same schema should fail"]]]
+                  (testing test-string
+                    (mt/with-all-users-data-perms-graph! {db-id {:data {:native :none, :schemas {(or schema-name "") schema-perms}}}}
+                      (if can-append?
+                        (is (some? (append-csv!)))
+                        (is (thrown-with-msg?
+                             clojure.lang.ExceptionInfo
+                             #"You don't have permissions to do that\."
+                             (append-csv!)))))))))))))))
+
+(deftest update-csv-block-perms-test
   (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
-    (testing "Appends should be blocked if the user has blocked data access to the database, unless they have native query editing perms too"
-      (upload-test/with-upload-table!
-        [table-a (upload-test/create-upload-table!)]
-        (let [db-id       (u/the-id (mt/db))
-              append-csv! #(upload-test/append-csv-with-defaults!
-                            :table-id (:id table-a)
-                            :user-id (mt/user->id :rasta))]
-          (testing "With blocked perms it should fail"
-            (mt/with-all-users-data-perms-graph! {db-id {:data {:native :none, :schemas :block}}}
-              (is (thrown-with-msg?
-                   clojure.lang.ExceptionInfo
-                   #"You don't have permissions to do that\."
-                   (append-csv!))))))))))
+    (doseq [action [:metabase.upload/append :metabase.upload/replace]]
+      (testing (format "We will block %s if the user has blocked data access to the database, even if they have native query editing"
+                       action)
+        (upload-test/with-upload-table!
+          [table-a (upload-test/create-upload-table!)]
+          (let [db-id       (u/the-id (mt/db))
+                append-csv! #(upload-test/update-csv-with-defaults!
+                              action
+                              :table-id (:id table-a)
+                              :user-id (mt/user->id :rasta))]
+            (doseq [native-permissions [:none :write]]
```

Comment:
I followed up on Noah's changes to this test in [this PR](https://github.com/metabase/metabase/pull/37473), and I'm pretty sure his changes were legit, we don't need to test `{:native :write}` perms here.

He just didn't update the `testing` description (it's correct with these changes).

See this block in notion:
https://www.notion.so/metabase/Split-data-access-and-query-builder-access-merge-GUI-and-SQL-f18185f2c2954127bdfcb2864ef5c13f?pvs=4#b3b035c280f546e3aedb8a393b27d4ce

> If a group has Blocked database access, both native query editing and query builder access must be No.

That means this is invalid state and we don't have to test it here.